python/ray/data/_internal/execution/interfaces.py

```diff
@@ -445,6 +453,17 @@ def incremental_resource_usage(self) -> ExecutionResources:
         """
         return ExecutionResources()
 
+    def notify_resource_usage(
+        self, input_queue_size: int, under_resource_limits: bool
+    ) -> None:
+        """Called periodically by the executor.
+
+        Args:
+            input_queue_size: The number of inputs queued outside this operator.
+            under_resource_limits: Whether this operator is under resource limits.
```

Comment:
```suggestion
            under_resource_limits: Whether all operators are under resource limits.
```

Reply:
this still indicates whether this particular operator is under resource limits right? Is Eric saying above that the executor is the own who knows about all operators' limits, but would also be the one to dictate whether this particular operator is under resource limits?

Reply:
Yeah, I think it's still about this particular operator. It's possible for some operators to be under limits but others to be above. For example, we count all downstream operator memory usage towards the "effective usage" of an operator.